DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -624,34 +624,70 @@ public async Task<ThreadQueryResult> ListJoinedPrivateArchivedThreadsAsync(DateT
     }
 
     /// <summary>
-    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="BadRequestException"/> error.
+    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="ArgumentException"/> error.
     /// </summary>
     /// <param name="messages">A collection of messages to delete.</param>
     /// <param name="reason">Reason for audit logs.</param>
-    /// <returns></returns>
-    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageMessages"/> permission.</exception>
-    /// <exception cref="NotFoundException">Thrown when the channel does not exist.</exception>
-    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
-    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async Task DeleteMessagesAsync(IEnumerable<DiscordMessage> messages, string reason = null)
+    /// <remarks>One api call per 100 requests</remarks>
+    public async Task<int> DeleteMessagesAsync(IReadOnlyList<DiscordMessage> messages, string reason = null)
     {
-        // don't enumerate more than once
-        ulong[] msgs = messages.Where(x => x.Channel.Id == this.Id).Select(x => x.Id).ToArray();
-        if (messages == null || !msgs.Any())
+        ArgumentNullException.ThrowIfNull(messages, nameof(messages));
+
+        List<ulong> messagesList = [];
+        foreach (DiscordMessage message in messages)
         {
-            throw new ArgumentException("You need to specify at least one message to delete.");
+            if (message.ChannelId != this.Id)
+            {
+                throw new ArgumentException(
+                    $"You cannot delete messages from channel {message.Channel.Name} through channel {this.Name}!.");
+            }
+            else if (message.Timestamp < DateTimeOffset.UtcNow.AddDays(-14))
+            {
+                throw new ArgumentException("You can only delete messages that are less than 14 days old.");
```

Comment:
i'm not sure how i feel about precluding these checks. at the very least it's a breaking change we should note.

Reply:
Its a known api limitation so i think we should check for it

Reply:
yes, but the current behaviour would be equivalent to checking per batch